pipeline/acquisition.py

```diff
@@ -0,0 +1,390 @@
+    -> Session
+    cell_id: varchar(36) # a string identifying the cell in which this intracellular recording is concerning
+    ---
+    cell_type: enum('excitatory','inhibitory','N/A')
+    -> reference.ActionLocation
+    -> reference.WholeCellDevice
+    """    
+  
+    
+@schema
+class IntracellularAcquisition(dj.Imported):
+    definition = """ # data pertain to intracellular recording
+    -> Cell
+    """     
+    
+    class MembranePotential(dj.Part):
+        definition = """
+        -> master
+        ---
+        membrane_potential: longblob    # Membrane potential recording at this cell
+        membrane_potential_wo_spike: longblob # membrane potential without spike data, derived from membrane potential recording    
+        membrane_potential_start_time: float # first timepoint of membrane potential recording
+        membrane_potential_sampling_rate: float # sampling rate of membrane potential recording
+        """
+        
+    class CurrentInjection(dj.Part):
+        definition = """
+        -> master
+        ---
+        current_injection: longblob
+        current_injection_start_time: float # first timepoint of current injection recording
+        current_injection_sampling_rate: float # sampling rate of current injection recording
+        """
+        
+    def make(self,key):
+        ############## Dataset #################
+        sess_data_dir = os.path.join('..','data','whole_cell_nwb2.0')
+                
+        # Get the Session definition from the keys of this session
+        animal_id = key['subject_id']
+        date_of_experiment = key['session_time']
+        
+        # Search the files in filenames to find a match for "this" session (based on key)
+        sess_data_file = utilities.find_session_matched_nwbfile(sess_data_dir, animal_id, date_of_experiment)
+        if sess_data_file is None: 
+            return
+        nwb = h5.File(os.path.join(sess_data_dir,sess_data_file), 'r')
+        
+        #  ============= Now read the data and start ingesting =============
+        self.insert1(key)
+        print('Insert intracellular data for: subject: {0} - date: {1} - cell: {2}'.format(key['subject_id'],key['session_time'],key['cell_id']))
+        # -- MembranePotential
+        key['membrane_potential'] = nwb['acquisition']['timeseries']['membrane_potential']['data'].value
+        key['membrane_potential_wo_spike'] = nwb['analysis']['Vm_wo_spikes']['membrane_potential_wo_spike']['data'].value
+        membrane_potential_time_stamps = nwb['acquisition']['timeseries']['membrane_potential']['timestamps'].value
+        key['membrane_potential_start_time'] = membrane_potential_time_stamps[0]
+        key['membrane_potential_sampling_rate'] = 1/np.mean(np.diff(membrane_potential_time_stamps))        
+        self.MembranePotential.insert1(key, ignore_extra_fields=True)
```

Comment:
I suggest the following implementation of inserts, which avoids the use of `ignore_extra_fields`. 
```python
        self.MembranePotential.insert1(dict(key, 
            membrane_potential=nwb['acquisition']['timeseries']['membrane_potential']['data'].value,
            membrane_potential_wo_spike=nwb['analysis']['Vm_wo_spikes']['membrane_potential_wo_spike']['data'].value,
            membrane_potential_time_stamps=nwb['acquisition']['timeseries']['membrane_potential']['timestamps'].value,
            membrane_potential_start_time=membrane_potential_time_stamps[0],
            membrane_potential_sampling_rate=1/np.mean(np.diff(membrane_potential_time_stamps))))